ui/src/services/utils.js

```diff
@@ -84,8 +84,19 @@ export const sortCapacity = createSelector(
       const sortedList = list
         .filter(item => k8sCapacityRegex.test(item?.[sortBy]))
         .map(item => {
-          const unit = item[sortBy].slice(-2);
-          const size = item[sortBy].substring(0, item[sortBy].length - 2);
+          /**
+           * item[sortBy].split(k8sCapacityRegex) will return an array of
+           * 4 items, only the 2nd and 3rd are useful.
+           *
+           * We might want to improve the regex to get rid of the useless empty
+           * string at the 1st and 4th place of the array later but right now
+           * just keep the k8s regex
+           *
+           * @example
+           * "1Gi" => ["", "1", "Gi", ""]
+           * "123" => ["", "123", "", ""]
+           */
+          const [empty, size, unit] = item[sortBy].split(k8sCapacityRegex);
```

Comment:
As discussed on our common Slack channel, using `match` instead of `split` would make for a clearer code and remove the need for the big comment above.
Especially if you use named capture group.

For the record, I propose that we simplify the regex used, by using `^(?<size>[1-9][0-9]*)(?<unit>[kKMGTP]i?)?`
And then (to be checked if the code below is correct, I'm not a JS guru ^^):

```javascript
const capacityRegex = /^(?<size>[1-9][0-9]*)(?<unit>[kKMGTP]i?)?/
[…]
const match= item[sortBy].match(capacityRegex).groups;
const size = match.size
const unit = match.unit
```
Or something like that.